pkg/aws/cloud.go

```
@@ -234,6 +219,78 @@ type defaultCloud struct {
 	wafRegional services.WAFRegional
 	shield      services.Shield
 	rgt         services.RGT
+
+	assumeRoleElbV2    map[string]services.ELBV2
+	awsClientsProvider provider.AWSClientsProvider
+	logger             logr.Logger
+}
+
+// returns ELBV2 client for the given assumeRoleArn, or the default ELBV2 client if assumeRoleArn is empty
+func (c *defaultCloud) GetAssumedRoleELBV2(ctx context.Context, assumeRoleArn string, externalId string) services.ELBV2 {
+
+	if assumeRoleArn == "" {
+		return c.elbv2
+	}
+
+	assumedRoleELBV2, exists := c.assumeRoleElbV2[assumeRoleArn]
+	if exists {
+		return assumedRoleELBV2
+	}
+	c.logger.Info("awsCloud", "method", "GetAssumedRoleELBV2", "AssumeRoleArn", assumeRoleArn, "externalId", externalId)
+
+	////////////////
+	existingAwsConfig, _ := c.awsClientsProvider.GetAWSConfig(ctx, "GetAWSConfigForIAMRoleImpersonation")
+
+	sourceAccount := sts.NewFromConfig(*existingAwsConfig)
+	response, err := sourceAccount.AssumeRole(ctx, &sts.AssumeRoleInput{
+		RoleArn:         aws.String(assumeRoleArn),
+		RoleSessionName: aws.String("aws-load-balancer-controller"),
+		ExternalId:      aws.String(externalId),
+	})
+	if err != nil {
+		log.Fatalf("Unable to assume target role, %v. Attempting to use default client", err)
```

It might make more sense to just bubble the error up rather than falling back to the default client. That way temporary issues don't result in the wrong account being used in the elb operations.